src/modules/ekf2/EKF/ekf.h

```diff
@@ -1132,88 +1125,194 @@ class Ekf final : public EstimatorInterface
 	bool _ev_q_error_initialized{false};
 #endif // CONFIG_EKF2_EXTERNAL_VISION
 
-	void resetEstimatorAidStatus(estimator_aid_source1d_s &status) const
+	void updateEstimatorAidStatusStateReset(estimator_aid_source1d_s &status) const
 	{
-		// only bother resetting if timestamp_sample is set
-		if (status.timestamp_sample != 0) {
-			status.timestamp_sample = 0;
+		status.time_last_fuse = _time_delayed_us;
 
-			// preserve status.time_last_fuse
+		status.innovation = 0.f;
+		status.innovation_filtered = 0.f;
+		status.innovation_variance = status.observation_variance;
 
-			status.observation = 0;
-			status.observation_variance = 0;
+		status.test_ratio = 0.f;
+		status.test_ratio_filtered = 0.f;
 
-			status.innovation = 0;
-			status.innovation_variance = 0;
-			status.test_ratio = INFINITY;
+		status.innovation_rejected = false;
+		status.fused = true;
+	}
 
-			status.innovation_rejected = true;
-			status.fused = false;
-		}
+	void updateEstimatorAidStatusStateReset(estimator_aid_source1d_s &status, const uint64_t &timestamp_sample,
+	                             const float observation, const float observation_variance) const
+	{
+		status.timestamp_sample = timestamp_sample;
+
+		status.observation = observation;
+		status.observation_variance = observation_variance;
+
+		updateEstimatorAidStatusStateReset(status);
 	}
 
-	template <typename T>
-	void resetEstimatorAidStatus(T &status) const
+	void updateEstimatorAidStatus(estimator_aid_source1d_s &status, const uint64_t &timestamp_sample,
+				      const float &observation, const float &observation_variance,
+				      const float &innovation, const float &innovation_variance,
+				      float innovation_gate = 1.f) const
 	{
-		// only bother resetting if timestamp_sample is set
-		if (status.timestamp_sample != 0) {
-			status.timestamp_sample = 0;
+		bool innovation_rejected = false;
+
+		const float test_ratio = sq(innovation) / (sq(innovation_gate) * innovation_variance);
+
+		if ((status.timestamp_sample > 0) && (timestamp_sample > status.timestamp_sample)) {
+
+			const float dt_s = math::constrain((timestamp_sample - status.timestamp_sample) * 1e-6f, 0.001f, 1.f);
 
-			// preserve status.time_last_fuse
+			static constexpr float tau = 0.1f;
+			const float alpha = math::constrain(dt_s / (dt_s + tau), 0.f, 1.f);
```

Comment:
It seemed a bit arbitrary, but starting with alpha 0.1 seems fine to start with. I'll update the branch.

Reply:
I didn't use AlphaFilter directly because I'm trying to lean on storing everything in the struct (.msg). I was actually starting to think capturing these commonalities in a class per aid source might help clean up a bit more.

Then we could standardized using the AlphaFilter, measuring data rate, etc and then at the end simply populate + publish EstimatorAidSrc, but not have to depend on it being the primary data storage.